api/src/schemas.py

```diff
@@ -250,14 +262,20 @@ def check_time_density(cls, values):
             raise ValueError("If is_periodic is false, time_density must be null")
         return values
 
-    # collection id must be all lowercase, with optional - delimiter
-    @validator("collection")
-    def check_id(cls, collection):
-        if not re.match(r"[a-z]+(?:-[a-z]+)*", collection):
-            raise ValueError(
-                "Invalid id - id must be all lowercase, with optional '-' delimiters"
-            )
-        return collection
+
+class DataType(str, enum.Enum):
+    cog = "cog"
+    zarr = "zarr"
+
+
+class COGDataset(Dataset):
+    spatial_extent: BboxExtent
+    temporal_extent: TemporalExtent
+    sample_files: List[str]  # unknown how this will work with CMR
+    data_type: Literal[DataType.cog]
+
+    class Config:
+        extra = Extra.allow
```

Comment:
I'm not sure about this flag - do we anticipate a case where extra values might be included?